### DISCUSSION TITLE
Multiple annotations per candle and annotation positioning

### DISCUSSION THREAD

**Discussion by yaro90** [reactivemarkets/react-financial-charts#552] [Ideas]
Hi, is it possible to create multiple svg annotations per candle at specific price levels or at least create a single annotation at a specific price level rather than just the top or bottom of the candle? It would be a really useful feature if this is not possible. Thanks

**Comment by markmcdowell** [comment]
yes you can add as many annotations on a single chart as you like, just add multiple `Annotate` components. You also define where to place the annotation in both axis.

See https://github.com/reactivemarkets/react-financial-charts/blob/master/packages/stories/src/features/annotated/Annotated.tsx for an example

**Reply by yaro90** [reply]
Thank you for your reply. I don't mean per chart, I mean per candle. I sort of see a way to achieve this now but its not entirely good for several reasons. This is how I thought this may be achieved:
* Pass through the data first and find the max number of annotations on a single candle.
* Loop for the max number of annotations creating the Annotate component with the appropriate props and when condition.

The reason I believe this is not entirely good is for performance as I think each candle will be checked for max number of annotations to be present on it and then only display those that are relevant. The second reason is the logic and code to achieve this is a little bit messy and not intuitive.

Please let me know your thoughts on the approach and if there is a better way to do it. Thanks
